src/output-plugin-syslog.c

```diff
@@ -0,0 +1,129 @@
+ *
+ * File-like output for logging:  syslog
+ */
+
+#ifndef OS_WIN32
+#include "suricata-plugin.h" /* errno.h, string.h, etc. */
+#include "conf.h"            /* ConfNode, etc. */
+#include "output.h"          /* DEFAULT_LOG_* */
+#include "util-syslog.h"
+
+SCPlugin *SyslogPluginRegister(void);
+
+#define DEFAULT_ALERT_SYSLOG_FACILITY_STR       "local0"
+#define DEFAULT_ALERT_SYSLOG_FACILITY           LOG_LOCAL0
+#define DEFAULT_ALERT_SYSLOG_LEVEL              LOG_INFO
+
+#define OUTPUT_NAME	"syslog"
+
+typedef struct Context_ {
+    int alert_syslog_level;
+} Context;
+
+static int SyslogPluginInit(ConfNode *conf, void **init_data)
+{
+    Context *context = SCCalloc(1, sizeof(Context));
+    if (context == NULL) {
+        SCLogError(SC_ERR_MEM_ALLOC, "Unable to allocate context for %s", OUTPUT_NAME);
+        return -1;
+    }
+	const char *facility_s = ConfNodeLookupChildValue(conf, "facility");
+	if (facility_s == NULL) {
+		facility_s = DEFAULT_ALERT_SYSLOG_FACILITY_STR;
+	}
+
+	int facility = SCMapEnumNameToValue(facility_s, SCSyslogGetFacilityMap());
+	if (facility == -1) {
+		SCLogWarning(SC_ERR_INVALID_ARGUMENT, "Invalid syslog facility: \"%s\","
+				" now using \"%s\" as syslog facility", facility_s,
+				DEFAULT_ALERT_SYSLOG_FACILITY_STR);
+		facility = DEFAULT_ALERT_SYSLOG_FACILITY;
+	}
+
+	const char *level_s = ConfNodeLookupChildValue(conf, "level");
+	if (level_s != NULL) {
+		int level = SCMapEnumNameToValue(level_s, SCSyslogGetLogLevelMap());
+		if (level != -1) {
+			context->alert_syslog_level = level;
+		}
+	}
+
+	const char *ident = ConfNodeLookupChildValue(conf, "identity");
+	/* if null we just pass that to openlog, which will then
+	 * figure it out by itself. */
+
+	openlog(ident, LOG_PID|LOG_NDELAY, facility);
+    *init_data = context;
+    return 0;
+}
+
+static int SyslogPluginWrite(const char *buffer, int buffer_len, void *init_data)
+{
+    Context *context = init_data;
+    syslog(context->alert_syslog_level, "%s",
+            (const char *)buffer);
+
+    return 0;
+}
+
+static void SyslogPluginDeInit(void *init_data)
+{
+    if (init_data) {
+        closelog();
+        SCFree(init_data);
+    }
+}
+
+static void SyslogPluginInitialize(void) {
+	SCPluginFileType *plugin_data = SCCalloc(1, sizeof(SCPluginFileType));
+
+	if (plugin_data == NULL) {
+		FatalError(SC_ERR_MEM_ALLOC, "Unable to allocate plugin memory for %s",
+				   OUTPUT_NAME);
+	}
+
+	plugin_data->name = OUTPUT_NAME;
+	plugin_data->Open = SyslogPluginInit;
+	plugin_data->Close = SyslogPluginDeInit;
+	plugin_data->Write = SyslogPluginWrite;
+	if (!SCPluginRegisterFileType(plugin_data)) {
+		FatalError(SC_ERR_PLUGIN, "Failed to register filetype plugin: %s",
+				   OUTPUT_NAME);
+	}
+}
+
+SCPlugin SyslogPluginRegistration = {
+    .name = OUTPUT_NAME,
+    .author = "OISF Dev",
+    .license = "GPLv2",
+    .Init = SyslogPluginInitialize
+};
+
+SCPlugin *SyslogPluginRegister(void)
+{
+	return &SyslogPluginRegistration;
+}
```

Comment:
I think we can skip this bit. This is only required for discovery of externally loadable modules. In this case We could go straight to `SyslogPluginInitialize`.

Reply:
Agreed.

Reply:
Well -- the syslog module must be discoverable ... currently, that happens by inclusion on the plugin list.

Reply:
Can't `SyslogInitialize` just be called at startup when we register everything else? The plugin registration is more of a hardcoded name we can use to discover the registration function in a dynamically loaded module, that step isn't really needed for internal modules.

Reply:
Updated with latest commit.